tests/regression_tests/benchmark/test_model_performance.py

```diff
@@ -0,0 +1,116 @@
+import os
+
+import pytest
+
+from ludwig.benchmarking.benchmark import benchmark
+from ludwig.utils.data_utils import load_json, load_yaml
+
+BENCHMARKING_CONFIG = """
+experiment_name: {experiment_name}
+hyperopt: false
+process_config_file_path: {process_config_file_path}
+export:
+  export_artifacts: true
+  export_base_path: {export_base_path}
+profiler:
+  enable: true
+  use_torch_profiler: false
+  logging_interval: 0.1
+experiments:
+  - dataset_name: {dataset}
+    config_path: {config_path}
+"""
+
+dataset_name_to_metric = {
```

Comment:
Dataclasses are always a good idea! Added them as well as the YAMLs.

The tests up until your comment were on the local backend. I switched to running these tests on the Ray backend. However, since time and resource tracking aren't supported in LudwigProfiler on the Ray backend yet (bug [here](https://github.com/ludwig-ai/ludwig/pull/2462#issuecomment-1260337734)), I only added these YAMLs and dataclasses for the model performance metric. The plan is to add time and resource usage later.